substrate/frame/identity/src/types.rs

```diff
@@ -234,24 +231,25 @@ impl<Balance: Encode + Decode + MaxEncodedLen + Copy + Clone + Debug + Eq + Part
 pub trait IdentityInformationProvider:
 	Encode + Decode + MaxEncodedLen + Clone + Debug + Eq + PartialEq + TypeInfo
 {
-	/// Type capable of representing all of the fields present in the identity information as bit
-	/// flags in `u64` format.
-	type IdentityField: Clone + Debug + Eq + PartialEq + TypeInfo + U64BitFlag;
+	/// Type capable of representing all of the fields present in the identity information.
+	type FieldsBitFlags: Encode
+		+ Decode
+		+ Clone
+		+ Debug
+		+ Default
+		+ Eq
+		+ PartialEq
+		+ TypeInfo
+		+ MaxEncodedLen;
```

Comment:
```suggestion
	/// Type capable of holding information on which fields are set.
	type FieldsInfo: sp_runtime::traits::Member + Default;
```

Reply:
Naming is hard and I still don't like my name. Maybe @joepetrowski has a better idea.

Reply:
Ahh just have seen: https://github.com/paritytech/polkadot-sdk/pull/2048#discussion_r1380637556

However, we should not include "BitFlags" in the name, because it could be everything (also may a vector for example).

Reply:
> However, we should not include "BitFlags" in the name, because it could be everything (also may a vector for example).

Yeah good point. `FieldsIdentifier`? This one is difficult to name.

Reply:
Renamed it to `FieldsIdentifier`. Maybe we'll come up with a better name in the future :grin: